src/main/java/uk/gov/indexer/ctserver/CTEntryLeaf.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.indexer.ctserver;
+
+import java.util.Arrays;
+import java.util.Base64;
+
+@SuppressWarnings({"FieldCanBeLocal", "unused"})
+public class CTEntryLeaf {
+    public final byte version;
+
+    public final byte merkleLeafType;
+
+    public final byte[] timetamp;
+
+    public final byte[] entryType;
+
+    public final byte[] contentLength;
+
+    public final byte[] payload;
+
+    public final byte[] sctExtension;
+
+    public CTEntryLeaf(String leafHash) {
+        byte[] decodeBytes = Base64.getDecoder().decode(leafHash);
+
+        int decodeBytesLength = decodeBytes.length;
+
+        //first byte is version
```

Comment:
check https://github.com/google/certificate-transparency/blob/master/cpp/proto/xjson_serializer.cc#L65. It explains that how all bytes of leaf hash are defined.